src/services/accounts/AccountsBalanceInfoService.ts

```diff
@@ -1,44 +1,84 @@
-import { BlockHash } from '@polkadot/types/interfaces';
+import { Vec } from '@polkadot/types';
+import {
+	AccountData,
+	BalanceLock,
+	BlockHash,
+} from '@polkadot/types/interfaces';
+import { BadRequest } from 'http-errors';
 import { IAccountBalanceInfo } from 'src/types/responses';
 
 import { AbstractService } from '../AbstractService';
 
 export class AccountsBalanceInfoService extends AbstractService {
 	/**
 	 * Fetch balance information for an account at a given block.
+	 * N.B. assumes all non native tokens are from ORML tokens pallet.
```

Comment:
Does this mean the code here assumes all non-native tokens implements `MultiCurrency`? If yes, maybe we should rephrase this to that effect: "Assumes all non-native tokens are implemented in a pallet that implements the `MultiCurrency` trait, e.g. the ORML tokens pallet"?

Reply:
I say this because the query is hardcoded to use a pallet called `tokens` (see `api.query.tokens.accounts.at`). So since we have to do querys specific to a storage item in a pallet (int this case `Accounts` in `tokens`), it is not generic to any trait. Although if the `MultiCurrency` trait exposed a way to query acocunt balance info without having to know the specific storage item that would be great.